src/main/fc/rc.c

```diff
@@ -537,6 +551,16 @@ static FAST_CODE void processRcSmoothingFilter(void)
             *dst = rxDataToSmooth[i];
         }
     }
+
+    // for ANGLE and HORIZON, smooth rcDeflection on pitch and roll
+    for (int axis = FD_ROLL; axis <= FD_YAW; axis++) {
+        if ((FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(HORIZON_MODE)) && rcSmoothingData.filterInitialized && axis < FD_YAW) {
+            rcDeflectionSmoothed[axis] = pt3FilterApply(&rcSmoothingData.filterDeflection[axis], rcDeflection[axis]);
+        } else {
+            rcDeflectionSmoothed[axis] = rcDeflection[axis];
+        }
+    }
+
```

Comment:
You can pull a part of the if statement out of the loop and save the check if axis < FD_YAW 🤓 :
```c
    // for ANGLE and HORIZON, smooth rcDeflection on pitch and roll
    if ((FLIGHT_MODE(ANGLE_MODE) || FLIGHT_MODE(HORIZON_MODE)) && rcSmoothingData.filterInitialized) {
        for (int axis = FD_ROLL; axis <= FD_PITCH; axis++) {
            rcDeflectionSmoothed[axis] = pt3FilterApply(&rcSmoothingData.filterDeflection[axis], rcDeflection[axis]);
        }
        rcDeflectionSmoothed[FD_YAW] = rcDeflection[FD_YAW];
    }
```